gotocompany/compass/v1beta1/service.proto

```diff
@@ -1146,6 +1172,50 @@ message DeleteAssetRequest {
 
 message DeleteAssetResponse {}
 
+message DeleteAssetsRequest {
+  string q = 1 [
+    (validate.rules).string.ignore_empty = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "filter by specific query"
+    }
+  ];
+  string q_fields = 2 [
+    (validate.rules).string.ignore_empty = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "filter by multiple query fields"
+    }
+  ];
+  string types = 3 [
+    (validate.rules).string.ignore_empty = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "filter by multiple types"
+    }
+  ];
+  string services = 4 [
+    (validate.rules).string.ignore_empty = true,
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "filter by multiple services"
+    }
+  ];
+  map<string, string> data = 5 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+    description: "query result based on a (nested) field of the data asset. the nested field is written with period separated field name. eg, \"data[name.entity]\""
+  }];
+  string duration_before_current = 6 [
+    (validate.rules).string.min_len = 1,
```

Comment:
should be `min_len` or `ignore_empty`?

Reply:
Currently, the idea is adding time comparing to Filter type in Compass, and in DeleteAssetsRequest, the Filter will be required, with duration_for_upper_threshold (duration_before_current before) is required (not empty), thats why I set min_len

Reply:
use required instead?
https://github.com/bufbuild/protoc-gen-validate?tab=readme-ov-file#durations

Reply:
updated 👍